### PR TITLE
Add other languages in examples list

### LINKED ISSUE #1246: Add example hyperlinks for other languages in Examples list in the Quickstart Section of README file
Currently, the Examples list in the [Quickstart Section](https://github.com/earthly/earthly#quick-start) of the README file contains hyperlinks to languages like Go, Java, Cpp, JS
It would be beneficial to add links to other languages such as C, Cobol, Python, Ruby, Rust, Scala to make the navigation easier.

@vladaionescu 
If you find this issue useful, I would be happy to work on this 😃

### PR DESCRIPTION
**Pull request by KKVANONYMOUS**
Resolves #1246 

I have added hyperlinks to other languages in the examples list and left a few as @rohitjmathew already started working on it in this [PR](https://github.com/earthly/earthly/pull/1249).  Although if you like, I can add the rest of the languages as well.
Also, I have sorted the languages to make the list more presentable.

@vladaionescu @alexcb Please review it and let me know.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,12 +108,16 @@ Here are some resources to get you started with Earthly
...
 * 🏁 [Getting started guide](https://docs.earthly.dev/guides/basics)
 * 👀 [Examples](./examples)
-  * [Go](./examples/go)
+  * [C](./examples/c)
+  * [Cobol](./examples/cobol)
+  * [C++](./examples/cpp)
   * [Java](./examples/java)
   * [JS](./examples/js)
-  * [C++](./examples/cpp)
+  * [Go](./examples/go)
```
Comment (alexcb) [inline README.md:116]: can you sort these alphabetically

File: README.md [hunk 2]
```diff
@@ -108,12 +108,16 @@ Here are some resources to get you started with Earthly
...
+  * [Scala](./examples/scala)
   * [Mono-repo](./examples/monorepo)
   * [Multi-repo](./examples/multirepo)
+
```
Comment (alexcb) [inline README.md:120]: nit: please remove this empty newline.

File: README.md (hunk 1 above)
Comment (KKVANONYMOUS) [reply README.md:114]: Sorry for the mistake. I have rectified it.

File: README.md (hunk 2 above)
Comment (KKVANONYMOUS) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (alexcb) [reply README.md:114]: can you sort all of the language examples.

Comment (KKVANONYMOUS) [reply README.md:114]: Just for clarification, you want me to add all the remaining language examples and then sort them accordingly.

Comment (alexcb) [reply README.md:114]: I'm just talking about the current list in this README. e.g. `C`, `C++`, `Cobol`, ...

Comment (KKVANONYMOUS) [reply README.md:114]: Is this ok now?

**Pull request merged**